addons/sourcemod/scripting/nd_discord_playerlist.sp

```diff
@@ -0,0 +1,192 @@
+    SaveTeamsTimer = CreateTimer(1.0, Timer_SaveTeams);
+}
+
+public Action Timer_SaveTeams(Handle timer)
+{
+    if(SavingTeams) {
+        SaveTeamsTimer = CreateTimer(1.0, Timer_SaveTeams);
+        return Plugin_Continue;
+    }
+    SaveTeamsTimer = null;
+
+    if(!IsValidHandle(DatabaseHandle)) {
+        LogAction(0, -1, "Database not connected");
+        return Plugin_Continue;
+    }
+
+    SavingTeams = true;
+
+    char query[1024];
+    FormatEx(query, sizeof(query), "TRUNCATE TABLE players;");
+    SQL_TQuery(DatabaseHandle, SaveTeams_Step1_RemovePlayers, query);
+
+    return Plugin_Continue;
+}
+
+void SaveTeams_Step1_RemovePlayers(Handle owner, Handle handle, const char[] error, any:data)
+{
+    if (error[0] != '\0')
+    {
+        LogAction(0, -1, "Removing players failed: %s", error);
+        SavingTeams = false;
+        return;
+    }
+
+    int[] consortium = new int[MaxClients];
+    int numConsortium = 0;
+    int[] empire = new int[MaxClients];
+    int numEmpire = 0;
+    int[] spectator = new int[MaxClients];
+    int numSpectator = 0;
+    for (int client = 1; client <= GetClientCount(); client++)
+    {
+        if (IsClientConnected(client))
+        {
+            if (GetClientTeam(client) == 2)
+            {
+                consortium[numConsortium] = client;
+                numConsortium++;
+            }
+            else if(GetClientTeam(client) == 3)
+            {
+                empire[numEmpire] = client;
+                numEmpire++;
+            }
+            else
+            {
+                spectator[numSpectator] = client;
+                numSpectator++;
+            }
+        }
+    }
+
+    // We're done if we have no players to add
+    if(numConsortium + numEmpire + numSpectator == 0) {
+        SavingTeams = false;
+        return;
+    }
+
+    char query[10000];
+    char queryvalues[10000];
+    char[][] queryparts = new char[numConsortium + numEmpire + numSpectator][1024];
+    int part = 0;
+    char name[MAX_NAME_LENGTH];
+    char escapedName[MAX_NAME_LENGTH];
+
+    for (int player = 0; player < numConsortium; player++)
+    {
+        GetClientName(consortium[player], name, sizeof(name));
+        SQL_EscapeString(DatabaseHandle, name, escapedName, sizeof(escapedName));
+        FormatEx(queryparts[part++], 1024, "('%s', '%s')", escapedName, "consortium");
+    }
+
+    for (int player = 0; player < numEmpire; player++)
+    {
+        GetClientName(empire[player], name, sizeof(name));
+        SQL_EscapeString(DatabaseHandle, name, escapedName, sizeof(escapedName));
+        FormatEx(queryparts[part++], 1024, "('%s', '%s')", escapedName, "empire");
+    }
+
+    for (int player = 0; player < numSpectator; player++)
+    {
+        GetClientName(spectator[player], name, sizeof(name));
+        SQL_EscapeString(DatabaseHandle, name, escapedName, sizeof(escapedName));
+        FormatEx(queryparts[part++], 1024, "('%s', '%s')", escapedName, "spectator");
+    }
```

Comment:
Please `GetClientName` on `OnClientPutInServer(int client)` and store in global 3D char array. This is too intensive. `char clientNames[MAXPLAYERS+1][MAX_NAME_LENGTH];`

Reply:
done